andela/tesla-ah
# #167193975 CRUD - category of articles Added

#### What does this PR do?
In this PR, 
- I implemented the endpoints that will enable the admin to create the categories, and to update it.
- I also added the endpoints which enable all users to be able to view all articles related to a specific category
#### Description of Task to be completed?
Having the following Endpoints working:
`POST /api/category`
`PUT /api/category/:id`
`DELETE /api/category/:id`
`GET /api/category/:categoryName/articles`
#### How should this be manually tested?
- clone the repository with this branch (**CRUD-category-167193975**)
- go to your terminal and then write the following command
  `cd tesla-ah`
- after that try to install all dependencies that are required for this project, simply by running the following command:
   `yarn install`  or `yarn`
- Then running the server, by running the below commands:
   1. `sequelize db:migrate` for this command make sure that the `sequelize-cli` package is installed globally
   2. `yarn dev`
   3.  After running the server, Open your postman, and then try to send for searching the article based on the provided tag, author's name, article's title or any keywords
- for running the tests, write the below command in your terminal
     `yarn test:local`
 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#167193975](https://www.pivotaltracker.com/n/projects/2350798/stories/167193975)
#### Screenshots (if appropriate)
N/A
#### Questions:
